### ISSUE TITLE
Using spritesheets for Emojis rather than fonts

### ISSUE THREAD

**Issue by koyuawsmbrtn** [tuskyapp/Tusky#703]
This approach would be better, because Tusky would support Emojis, that aren't supported by the Android operating system. It would give users a more streamlined experience across the web version of Mastodon and their various Android devices running Tusky.

* * * *
- Tusky Version: 2.0+
- Android Version: any
- Android Device: any
- Mastodon instance (if applicable): any

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by charlag** [comment]
I may be mistaken but I think the approach we use now *should* allow you to see emojis which are not supported by your system. Could you specify your Android version pls?

**Comment by connyduck** [comment]
Yes our approach should definitely also replace emojis not supported by the platform.
Could you post a link to a status that does nor work for you & a screenshot of it please?
cc @C1710

**Comment by C1710** [comment]
Well, at least it supports _all_ emojis that are included in the latest Emoji 11 standard as well as some custom ZWJ-sequences (these sequences are only available for Blobmoji).  
Additional characters have to be included in the fonts, but as I mentioned both Twemoji and Blobmoji already support _all_ official emoji characters (maybe with the exception of some flags and some symbols which aren't rendered as emojis).

**Comment by C1710** [comment]
Anyhow I got some issues with ZWJ sequences not working on devices with KitKat. But this shouldn't affect a larger user group (and especially not the Tusky users, since the support for Lollipop and lower is currently disabled).

**Comment by C1710** [comment]
Here's an example of all the Emoji 11 emojis:  
https://mastodon.xyz/@C1710/100264999957588974  
If any of them are not visible, it's just another bug

**Comment by koyuawsmbrtn** [comment]
This is how it looks like on Android 6.0:

![image](https://user-images.githubusercontent.com/40538332/42185089-029978ae-7e48-11e8-8232-7e1c64b9ba80.png)

**Comment by C1710** [comment]
Well, that's definetly not the intended behavior... Which emoji pack do you use?

**Comment by koyuawsmbrtn** [comment]
Twemoji

**Comment by C1710** [comment]
So it's another bug with the two EmojiCompat fonts...  
My next assumption would be that the script that generates the fonts isn't made for more emojis than the ones available in Emoji 5.0, but that's just another wild guess.  
I'll have to check it again in the next few days, but i think I'll need some sleep now.  
But this really gets frustrating....  
(It's actually not (directly) the fault of the library but rather of the script that creates the EmojiCompat fonts, since the fonts themselves are definetly compatible with Android 4.4+)

**Comment by C1710** [comment]
Just for the records: The problem was, that older Android versions seem to have trouble with (this issue)[https://github.com/googlei18n/noto-emoji/issues/19].  
So in the end there was no issue with the fonts but with the Android OS not accepting the correct version.

**Comment by C1710** [comment]
@connyduck I guess you can close this issue 😅 (Fixed in [Pullrequest #5 to tuskyapp.github.io](https://github.com/tuskyapp/tuskyapp.github.io/pull/5))
